planner/core/exhaust_physical_plans.go

```diff
@@ -1657,7 +1657,7 @@ func (p *LogicalJoin) tryToGetIndexJoin(prop *property.PhysicalProperty) (indexJ
 			case hasINLMJHint:
 				errMsg = "Optimizer Hint INL_MERGE_JOIN is inapplicable"
 			}
-			if p.hintInfo != nil {
+			if p.hintInfo != nil && p.preferJoinType > 0 {
```

Comment:
Why change this?

Reply:
Previous, `p.hintInfo` was used to check whether we have a join method hint. It means, that if `p.hintInfo != nil`, there is a join method hint like `hash join` hint. But now, `p.hintInfo != nil` also can means there is a leading hint. So we need to check the `p.preferJoinType` to see whether the hint is join method or join order hint.